tests/integration/select-component-test.js

```diff
@@ -672,7 +673,62 @@ test('Dropdown does not open on key event when select is disabled', function(ass
   selectElement.focus();
   pressDownArrow(selectElement);
 
-  andThen(() => {
+  return andThen(() => {
     assert.ok(isNotPresent('.ember-select-results', selectElement), 'The dropdown does not open');
   });
 });
+
+test('Collapsed group can be expanded, collapsed', function(assert) {
+  function data() {
+    return [
+      {name: 'Sparrow', sound: 'Squawk'},
+      {name: 'Crow', sound: 'Squawk'},
+      {name: 'Dog', sound: 'bark'},
+      {name: 'Wolf', sound: 'Bark'},
+      {name: 'Sea Lion', sound: 'Bark'}
+    ];
+  }
+  function renderedItems(selectElement) {
+    var resultItems = find('.ember-select-result-item', selectElement);
+    return resultItems.toArray().map(i => i.textContent.trim())
+  }
+
+  select = this.subject({
+    content: data(),
+    optionLabelPath: 'name',
+    optionValuePath: 'name',
+    optionGroupPath: 'sound',
+    isGroupHeaderCollapsible: true,
+    collapsedGroupHeaders: Ember.A(['Squawk', 'bark'])
+  });
+  this.append();
+
+  var selectElement = select.$();
+  openDropdown(selectElement);
+
+  andThen(() => {
+    assert.deepEqual(
+      renderedItems(selectElement),
+      ['Bark', 'Sea Lion', 'Wolf', 'Squawk', 'bark'],
+      'precond - rendered content includes groups'
+    );
+    click('li:contains(bark) div');
+  });
+
+  andThen(() => {
+    assert.deepEqual(
+      renderedItems(selectElement),
+      ['Bark', 'Sea Lion', 'Wolf', 'Squawk', 'bark', 'Dog'],
+      'bark expanded to render Dog'
+    );
+    click('li:contains(Bark) div');
+  });
+
+  return andThen(() => {
+    assert.deepEqual(
```

Comment:
Is it needed to wrap the assertions in `andThen`?

Reply:
Yeah, the `click` may have resulted in `rAF` frames from vertical collection, since this is a select (which uses vertical collection). They might not be hard required in this case, but in general the this test should be async-friendly, permitting `rAF` and such if it was needed.